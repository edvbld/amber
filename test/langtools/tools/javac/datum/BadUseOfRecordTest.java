/*
 * @test /nodynamiccopyright/
 * @summary check that record is not used as a type name
 * @compile/fail/ref=BadUseOfRecordTest.out -XDrawDiagnostics BadUseOfRecordTest.java
 */

class BadUseOfRecordTest {

    interface record {
        void m();
    }

    class record {}

    enum record {A, B}

    record record(int x);
}
