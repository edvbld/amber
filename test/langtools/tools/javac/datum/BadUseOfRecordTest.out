BadUseOfRecordTest.java:9:15: compiler.err.record.not.allowed: record
BadUseOfRecordTest.java:13:11: compiler.err.record.not.allowed: record
BadUseOfRecordTest.java:15:10: compiler.err.record.not.allowed: record
BadUseOfRecordTest.java:17:12: compiler.err.record.not.allowed: record
4 errors
